quantum/rgb_matrix/rgb_matrix.c

```diff
@@ -426,12 +428,21 @@ void rgb_matrix_task(void) {
 
     uint8_t effect = suspend_backlight || !rgb_matrix_config.enable ? 0 : rgb_matrix_config.mode;
 
+    // only for static modes or upon initialisation
+    if (rgb_matrix_config.mode > 4 || rgb_effect_params.init) {
+        render_update = true;
+    }
```

Comment:
I think this could be done better without checking the mode is >4 but unsure of how